### ISSUE TITLE
Error by default

### ISSUE THREAD

**Issue by serejahh** [realm/SwiftLint#834]
Hey. Thanks for the awesome tool

I want to enforce using SwiftLint and mark all SwiftLint warning as errors. Now I have to write:

```
line_length: 
  warning: 120
  error: 120
```

Can I set some option like in Xcode `treat warnings as errors` for all SwiftLint warnings?

**Comment by BluMist** [comment]
This can be achieved by using warning_threshold: 0 in your .yml file

**Comment by raphaelcruzeiro** [comment]
Any way to treat only SwiftLint warnings as errors? `warning_threshold: 0` will treat external warnings as errors as well.

**Comment by brynjar** [comment]
So it seems that if you want to use the default rules (with some disabled possibly) you have to list every rule separately, e.g.

```
control_statement:
  severity: error
```

to get them all to produce errors. I don't want to use xcode settings.

**Comment by Antondomashnev** [comment]
Did anybody try `warning_threshold: 0`? I have an Xcode build setting `treat warnings as errors`, so the build has no warnings/error, but the swiftlint fails. It works only if I set `warning_threshold: 1`.

**Comment by jpsim** [comment]
> Can I set some option like in Xcode `treat warnings as errors` for all SwiftLint warnings?

Yes, that's the `--strict` option to the `lint` command:

```shell
$ swiftlint help lint
Print lint warnings and errors (default command)

...

[--strict]
	fail on warnings
```

**Comment by mxcl** [comment]
As an FYI for people discovering this thread, `--strict` does not convert warnings to errors, it changes the exit code for lints that only contains warnings to non-zero, which causes Xcode to say that the build phase failed, but doesn't change any of the warnings into errors.

**Comment by mrkd** [comment]
> Did anybody try `warning_threshold: 0`? I have an Xcode build setting `treat warnings as errors`, so the build has no warnings/error, but the swiftlint fails. It works only if I set `warning_threshold: 1`.

I ran into the same problem and opened an issue:
https://github.com/realm/SwiftLint/issues/2403

**Comment by arielelkin** [comment]
for those integrating SwiftLint via Cocoapods, update your SwiftLint build phase to run this script:

`${PODS_ROOT}/SwiftLint/swiftlint --strict`
